airflow/providers/amazon/aws/hooks/athena.py

```diff
@@ -253,7 +253,7 @@ def poll_query_status(
         try:
             wait(
                 waiter=self.get_waiter("query_complete"),
-                waiter_delay=sleep_time or self.sleep_time,
+                waiter_delay=self.sleep_time if sleep_time is None else sleep_time,
```

Comment:
I'll leave this up to you, but it seems easier to read as 

```suggestion
                waiter_delay=sleep_time if sleep_time else self.sleep_time,
```

which should also boil down further to just
 
```suggestion
                waiter_delay=sleep_time or self.sleep_time,
```
![image](https://github.com/apache/airflow/assets/1920178/e478bae7-4179-4ef9-961b-959c97e95be3)

Reply:
I think I wrote it this way to allow `sleep_time = 0`. `is None` will keep the zero value, but `or` will replace it with the alternative, and being able to specify a sleep time of 0 is useful in tests.

Reply:
didn't I put a self-review comment about that ? 🤔